packages/@react-stately/table/src/TableUtils.ts

```diff
@@ -133,42 +65,194 @@ export interface IColumn {
+  // 9.7.4
+  // 9.7.4.a
+  while (hasNonFrozenItems) {
+    // 9.7.4.b
+    /**
+     * Calculate the remaining free space as for initial free space,
+     * above (9.7.3). If the sum of the unfrozen flex items’ flex factors is
+     * less than one, multiply the initial free space by this sum (of flex factors).
+     * If the magnitude of this value is less than the magnitude of
+     * the remaining free space, use this as the remaining free space.
+     */
+    let usedWidth = 0;
+    let flexFactors = 0;
+    flexItems.forEach(item => {
+      if (item.frozen) {
+        usedWidth += item.targetMainSize;
+      } else {
+        usedWidth += item.baseSize;
+        flexFactors += item.flex;
+      }
+    });
+
+    remainingFreeSpace = availableWidth - usedWidth;
+    // we only support integer FR's, and because of hasNonFrozenItems, we know that flexFactors > 0
+    // so no need to check for flexFactors < 1
+    // 9.7.4.c
+    /**
+     * If the remaining free space is zero
+     * - Do nothing.
+     * Else // remember, we're always in grow mode
+     * - Find the ratio of the item’s flex grow factor to the
+     * sum of the flex grow factors of all unfrozen items on
+     * the line. Set the item’s target main size to its flex
+     * base size plus a fraction of the remaining free space
+     * proportional to the ratio.
+     */
+    if (remainingFreeSpace > 0) {
+      flexItems.forEach((item) => {
+        if (!item.frozen) {
+          let ratio = item.flex / flexFactors;
+          item.targetMainSize = item.baseSize + (ratio * remainingFreeSpace);
+        }
+      });
+    }
+
+    // 9.7.4.d
+    /**
+     * Fix min/max violations. Clamp each non-frozen item’s
+     * target main size by its used min and max main sizes
+     * and floor its content-box size at zero. If the item’s
+     * target main size was made smaller by this, it’s a max
+     * violation. If the item’s target main size was made
+     * larger by this, it’s a min violation.
+     */
+    let totalViolation = 0;
+    flexItems.forEach(item => {
+      item.violation = 0;
+      if (!item.frozen) {
+        let {min, max, targetMainSize} = item;
+        item.targetMainSize = Math.max(min, Math.min(targetMainSize, max));
+
+        item.violation = item.targetMainSize - targetMainSize;
+        totalViolation += item.violation;
+      }
+    });
+
+    // 9.7.4.e
+    /**
+     * Freeze over-flexed items. The total violation is the
+     * sum of the adjustments from the previous step
+     * ∑(clamped size - unclamped size). If the total violation is:
+     * Zero
+     * - Freeze all items.
+     *
+     * Positive
+     * - Freeze all the items with min violations.
+     *
+     * Negative
+     * - Freeze all the items with max violations.
+     */
+    hasNonFrozenItems = false;
+    flexItems.forEach(item => {
+      if (totalViolation === 0 || Math.sign(totalViolation) === Math.sign(item.violation)) {
+        item.frozen = true;
+      } else {
```

Comment:
Shouldn't this check be 
```suggestion
      } else if (!item.frozen) {
```
so that we avoid running through the loop again even if all items are frozen.

From my understanding, it is possible for a `flexItem` to be `frozen` (and therefore have its `violation` be set to 0) and thus have the `if` about fail if totalViolation is non zero. Alternatively, the inside of the forEach could just have a   wrapping `if (!item.frozen) {` or a `filter` applied so we only iterate over non-frozen items.

Reply:
filter loops over everything anyways to construct a new array, this is no more expensive. good catch on the !item.frozen check though